File: ESTRUTURAS/1281/main.dart
import 'dart:io';

Map<String, double> getProducts() {
  Map<String, double> products = {};
  int m = int.parse(stdin.readLineSync()!);
  while (m-- > 0) {
    var [product, price] = stdin.readLineSync()!.split(' ');
    products[product] = double.parse(price);
  }
  return products;
}

double getTotal(Map<String, double> products) {
  double total = 0;
  int p = int.parse(stdin.readLineSync()!);
  while (p-- > 0) {
    var [product, qty] = stdin.readLineSync()!.split(' ');
    total += products[product]! * int.parse(qty);
  }
  return total;
}

void main() {
  int n = int.parse(stdin.readLineSync()!);
  while (n-- > 0) {
    Map<String, double> products = getProducts();
    double total = getTotal(products);
    print('R\$ ${total.toStringAsFixed(2)}');
  }
}

File: ESTRUTURAS/1281/main.rb
# frozen_string_literal:true

def get_products
  products = {}
  m = gets.to_i
  m.times do
    product, price = gets.split
    products[product] = price.to_f
  end
  products
end

def get_total(products)
  total = 0
  p = gets.to_i
  p.times do
    product, qty = gets.split
    total += products[product] * qty.to_i
  end
  total
end

n = gets.to_i
n.times do
  products = get_products
  total = get_total(products)
  puts format('R$ %.2f', total)
end
